### ISSUE TITLE
Check box automatically uncheck when we update same checkbox more than 1 times

### ISSUE THREAD

**Issue by remove158** [pymupdf/PyMuPDF#2391]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
Check box automatically uncheck when we update same checkbox more than 1 times

## To Reproduce (mandatory)
```py
import fitz

doc = fitz.Document("32.pdf")

# its work when we update first-time
for page in doc:
    widgets = page.widgets()
    for field in widgets:
        if field.field_type == fitz.PDF_WIDGET_TYPE_CHECKBOX:
            field.field_value = True
        field.update() 

# now after we set to read-only the checked symbol that we check before is gone.
for page in doc:
    widgets = page.widgets()
    for field in widgets:
        if field.field_type == fitz.PDF_WIDGET_TYPE_CHECKBOX:
            field.field_flags |= fitz.PDF_FIELD_IS_READ_ONLY
        field.update()  

doc.save("output.pdf")
```

## Expected behavior (optional)
If we update only one time every check box should be checked (works)
If we update more than 1 time. checkbox should update base on last change same as text-field behavior.  (didn't work)
- if last change is checked. next time if I update read-only true. It should be checked+read-only

## Your configuration (mandatory)
 - Operating system,  MacOS latest
 - Python version, 3.11
 - PyMuPDF version ^1.22.2

**Comment by JorjMcKie** [comment]
Found the problem and already developed a solution:

The problem is that for checkboxes being checked, a value of "Yes" is returned- not `True`. The current update logic however interprets any value not equal to `True`as `False`.

So the easy fix is simply also allowing "Yes" as an alias for "checked".

**Comment by JorjMcKie** [comment]
As an obvious circumvention in your read-only loop, you can check for value "Yes" and overwrite the field value with `True` in that case.

**Comment by remove158** [comment]
Okay i see, for now I just use to work-around.
```py 
field.field_value == "/Yes" or field.field_value == "Yes" or field.field_value == True
```

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.22.5.
